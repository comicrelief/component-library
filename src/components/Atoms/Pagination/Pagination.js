import React from 'react';
import PropTypes from 'prop-types';
import PaginationList from './Components/PaginationList';
import PaginationItem from './Components/PaginationItem';
import { getPages } from './Utils/PaginationCalculator';

const Pagination = ({
  PageComponent,
  totalPages,
  maxPages,
  currentPage,
  showFirst,
  firstLabel,
  firstAriaLabel,
  showPrevious,
  previousLabel,
  previousAriaLabel,
  showNext,
  nextLabel,
  nextAriaLabel,
  showLast,
  lastLabel,
  lastAriaLabel,
  getPageLabel,
  getPageAriaLabel,
  onSelect,
  createURL,
  className
}) => {
  const getItem = (pageClassName, label, ariaLabel, value, disabled) => {
    return {
      key: `${pageClassName}.${value}`,
      pageClassName,
      disabled,
      selected: currentPage === value,
      label,
      value,
      ariaLabel
    };
  };
  let items = [];
  if (showFirst) {
    items.push(
      getItem(
        'item--firstPage',
        firstLabel,
        firstAriaLabel,
        1,
        currentPage === 1
      )
    );
  }
  if (showPrevious) {
    items.push(
      getItem(
        'item--previousPage',
        previousLabel,
        previousAriaLabel,
        currentPage - 1,
        currentPage === 1
      )
    );
  }

  items = items.concat(
    getPages(currentPage, maxPages, totalPages).map(value =>
      getItem(
        'item--page',
        getPageLabel(value),
        getPageAriaLabel(value),
        value,
        currentPage === value
      )
    )
  );
  if (showNext) {
    items.push(
      getItem(
        'item--nextPage',
        nextLabel,
        nextAriaLabel,
        currentPage + 1,
        currentPage === maxPages || maxPages <= 1
      )
    );
  }
  if (showLast) {
    items.push(
      getItem(
        'item--lastPage',
        lastLabel,
        lastAriaLabel,
        maxPages,
        currentPage === maxPages || maxPages <= 1
      )
    );
  }

  return (
    <PaginationList
      className={className}
      pages={items}
      PageComponent={PageComponent}
      createURL={createURL}
      onSelect={onSelect}
    />
  );
};
Pagination.propTypes = {
  totalPages: PropTypes.number.isRequired, // total pages
  currentPage: PropTypes.number.isRequired,
  createURL: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  PageComponent: PropTypes.func,
  showFirst: PropTypes.bool,
  showPrevious: PropTypes.bool,
  showNext: PropTypes.bool,
  showLast: PropTypes.bool,
  maxPages: PropTypes.number, // allowed max number of pages
  className: PropTypes.string,
  previousLabel: PropTypes.string,
  nextLabel: PropTypes.string,
  firstLabel: PropTypes.string,
  lastLabel: PropTypes.string,
  getPageLabel: PropTypes.func,
  previousAriaLabel: PropTypes.string,
  nextAriaLabel: PropTypes.string,
  firstAriaLabel: PropTypes.string,
  lastAriaLabel: PropTypes.string,
  getPageAriaLabel: PropTypes.func
};

Pagination.defaultProps = {
  PageComponent: PaginationItem,
  showFirst: true,
  showPrevious: true,
  showNext: true,
  showLast: true,
  maxPages: 5,
  className: '',
  previousLabel: '‹',
  nextLabel: '›',
  firstLabel: '«',
  lastLabel: '»',
  getPageLabel: currentPage => currentPage.toString(),
  previousAriaLabel: 'Previous page',
  nextAriaLabel: 'Next page',
  firstAriaLabel: 'First page',
  lastAriaLabel: 'Last page',
  getPageAriaLabel: currentPage => `Page ${currentPage.toString()}`
};
export default Pagination;
