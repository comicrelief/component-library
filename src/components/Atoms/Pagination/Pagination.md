# Pagination

Minimalist.

````js
<Pagination
  maxPages={5}
  totalPages={5}
  currentPage={2}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={false}
  showPrevious={false}
  showNext={false}
  showLast={false}
/>;```

Full size.

```js
<Pagination
  maxPages={5}
  totalPages={5}
  currentPage={2}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
/>;
````

Custom labels.

```js
<Pagination
  maxPages={5}
  totalPages={5}
  currentPage={2}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  previousLabel="Previous"
  nextLabel="Next"
  firstLabel="First"
  lastLabel="Last"
  getPageLabel={value => `Page ${value}`}
/>
```
