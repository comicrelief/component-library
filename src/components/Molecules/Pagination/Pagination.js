import React from 'react';
import PropTypes from 'prop-types';
import StyledPaginationList from './Components/PaginationList.style';
import StyledPaginationItem from './Components/PaginationItem.style';
import { getPages } from './Utils/PaginationCalculator';

/**
 * Customizable Pagination component
 */
const Pagination = ({
  PageComponent,
  totalPages,
  maxPages,
  currentPage,
  showFirst,
  firstLabel,
  firstAriaLabel,
  showPrevious,
  previousLabel,
  previousAriaLabel,
  showMore,
  moreLabel,
  moreAriaLabel,
  showNext,
  nextLabel,
  nextAriaLabel,
  showLast,
  lastLabel,
  lastAriaLabel,
  getPageLabel,
  getPageAriaLabel,
  onSelect,
  createURL,
  className,
  pageComponentProps
}) => {
  const getItem = (pageClassName, label, ariaLabel, value, disabled) => {
    return {
      key: `${pageClassName}.${value}`,
      pageClassName,
      disabled,
      selected: currentPage === value,
      label,
      value,
      ariaLabel
    };
  };
  let items = [];
  if (showFirst) {
    items.push(
      getItem('firstPage', firstLabel, firstAriaLabel, 1, currentPage === 1)
    );
  }
  if (showPrevious) {
    items.push(
      getItem(
        'previousPage',
        previousLabel,
        previousAriaLabel,
        currentPage - 1,
        currentPage === 1
      )
    );
  }

  const pages = getPages(currentPage, maxPages, totalPages);
  if (showMore && pages[0] > 1) {
    items.push(getItem('morePagesBefore', moreLabel, moreAriaLabel, 0, true));
  }
  items = items.concat(
    pages.map(value =>
      getItem(
        'page',
        getPageLabel(value),
        getPageAriaLabel(value),
        value,
        false
      )
    )
  );
  if (showMore && pages[pages.length - 1] < totalPages) {
    items.push(getItem('morePagesAfter', moreLabel, moreAriaLabel, 0, true));
  }
  if (showNext) {
    items.push(
      getItem(
        'nextPage',
        nextLabel,
        nextAriaLabel,
        currentPage + 1,
        currentPage === totalPages || totalPages === 1
      )
    );
  }
  if (showLast) {
    items.push(
      getItem(
        'lastPage',
        lastLabel,
        lastAriaLabel,
        totalPages,
        currentPage === totalPages || totalPages === 1
      )
    );
  }

  return (
    <StyledPaginationList
      className={className}
      pages={items}
      PageComponent={PageComponent}
      pageComponentProps={pageComponentProps}
      createURL={createURL}
      onSelect={onSelect}
    />
  );
};
Pagination.propTypes = {
  /** Total number of pages */
  totalPages: PropTypes.number.isRequired,
  /** Current select page */
  currentPage: PropTypes.number.isRequired,
  /** Create each page url, page number is available as an argument */
  createURL: PropTypes.func.isRequired,
  /** Select a page, event and page number are provided as arguments respectively */
  onSelect: PropTypes.func.isRequired,
  /** Component to use to render each page */
  PageComponent: PropTypes.elementType,
  /** Props for component used to render each page */
  pageComponentProps: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.string,
      PropTypes.number,
      PropTypes.func
    ])
  ),
  /** Show first page link */
  showFirst: PropTypes.bool,
  /** Show previous page link */
  showPrevious: PropTypes.bool,
  /** Show next page link */
  showNext: PropTypes.bool,
  /** Show last page link */
  showLast: PropTypes.bool,
  /** Show that there are more pages */
  showMore: PropTypes.bool,
  /** Max allowed number of pages */
  maxPages: PropTypes.number,
  /** Main pagination container class */
  className: PropTypes.string,
  /** Previous page label */
  previousLabel: PropTypes.string,
  /** Next page label */
  nextLabel: PropTypes.string,
  /** First page label */
  firstLabel: PropTypes.string,
  /** Last page label */
  lastLabel: PropTypes.string,
  /** More pages -other than displayed ones- label */
  moreLabel: PropTypes.string,
  /** Generate page label, page number is available as an argument */
  getPageLabel: PropTypes.func,
  /** Previous page aria label */
  previousAriaLabel: PropTypes.string,
  /** Next page aria label */
  nextAriaLabel: PropTypes.string,
  /** First page aria label */
  firstAriaLabel: PropTypes.string,
  /** Last page aria label */
  lastAriaLabel: PropTypes.string,
  /** More pages -other than displayed ones- aria label */
  moreAriaLabel: PropTypes.string,
  /** Generate page aria label, page number is available as an argument */
  getPageAriaLabel: PropTypes.func
};

Pagination.defaultProps = {
  PageComponent: StyledPaginationItem,
  pageComponentProps: { color: 'red' },
  showFirst: true,
  showPrevious: true,
  showNext: true,
  showLast: true,
  showMore: true,
  maxPages: 5,
  className: '',
  previousLabel: '‹',
  nextLabel: '›',
  firstLabel: '«',
  lastLabel: '»',
  moreLabel: '...',
  getPageLabel: currentPage => currentPage.toString(),
  previousAriaLabel: 'Previous page',
  nextAriaLabel: 'Next page',
  firstAriaLabel: 'First page',
  lastAriaLabel: 'Last page',
  moreAriaLabel: 'More pages',
  getPageAriaLabel: currentPage => `Page ${currentPage}`
};
export default Pagination;
