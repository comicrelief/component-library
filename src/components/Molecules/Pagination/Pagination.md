# Pagination

## Minimalist

```js
<Pagination
  maxPages={6}
  totalPages={6}
  currentPage={2}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={false}
  showPrevious={false}
  showNext={false}
  showLast={false}
/>

<Pagination
  maxPages={6}
  totalPages={6}
  currentPage={4}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={false}
  showPrevious={false}
  showNext={false}
  showLast={false}
  pageComponentProps={{color: 'black'}}
/>
```

## Full size

```js
<Pagination
  maxPages={5}
  totalPages={10}
  currentPage={1}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  showMore={true}
/>

<Pagination
  maxPages={5}
  totalPages={10}
  currentPage={3}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  showMore={true}
  pageComponentProps={{color: 'purple'}}
/>

<Pagination
  maxPages={5}
  totalPages={10}
  currentPage={5}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  showMore={true}
  pageComponentProps={{color: 'blue'}}
/>

<Pagination
  maxPages={5}
  totalPages={10}
  currentPage={10}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  showMore={true}
  pageComponentProps={{color: 'green'}}
/>
```

## Custom labels

```js
<Pagination
  maxPages={5}
  totalPages={20}
  currentPage={10}
  createURL={value => `example.com/page/${value}`}
  onSelect={(event, value) => {
    event.preventDefault();
    console.log(value);
  }}
  showFirst={true}
  showPrevious={true}
  showNext={true}
  showLast={true}
  showMore={true}
  previousLabel="Previous"
  nextLabel="Next"
  firstLabel="First"
  lastLabel="Last"
  moreLabel="More"
  getPageLabel={value => `Page ${value}`}
/>
```
