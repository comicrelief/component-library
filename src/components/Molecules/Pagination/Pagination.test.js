import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import Pagination from './Pagination';

it('renders correctly in minimalist form', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={5}
      totalPages={5}
      currentPage={3}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst={false}
      showPrevious={false}
      showNext={false}
      showLast={false}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly in full size with max pages equal to total pages', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={5}
      totalPages={5}
      currentPage={3}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst
      showPrevious
      showNext
      showLast
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly in full size with max pages more than total pages', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={10}
      totalPages={6}
      currentPage={4}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst
      showPrevious
      showNext
      showLast
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly in full size with max pages less than total pages', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={6}
      totalPages={10}
      currentPage={2}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst
      showPrevious
      showNext
      showLast
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly in full size with first page selected', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={5}
      totalPages={5}
      currentPage={1}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst
      showPrevious
      showNext
      showLast
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with custom labels', () => {
  const tree = renderWithTheme(
    <Pagination
      maxPages={5}
      totalPages={5}
      currentPage={3}
      createURL={value => `example.com/page/${value}`}
      onSelect={value => value}
      showFirst
      showPrevious
      showNext
      showLast
      previousLabel="Previous"
      nextLabel="Next"
      firstLabel="First"
      lastLabel="Last"
      getPageLabel={value => `Page ${value}`}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
