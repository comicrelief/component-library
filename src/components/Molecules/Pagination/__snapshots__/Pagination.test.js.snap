// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly in full size with max pages equal to total pages 1`] = `
.c0 {
  display: inline-block;
  padding: 16px 30px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 30px;
  cursor: point;
  background-color: #f04257;
  background: #f04257;
  color: #FFFFFF;
}

.c0:hover {
  background: #00beca;
}

<div
  className=""
>
  <ul
    className="list"
  >
    <li
      className="item   item--firstPage"
    >
      <a
        aria-label="First page"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        «
      </a>
    </li>
    <li
      className="item   item--previousPage"
    >
      <a
        aria-label="Previous page"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        ‹
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 1"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        1
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 2"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        2
      </a>
    </li>
    <li
      className="item  item--disabled item--page"
    >
      <span
        className="link"
      >
        3
      </span>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 4"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        4
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 5"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        5
      </a>
    </li>
    <li
      className="item   item--nextPage"
    >
      <a
        aria-label="Next page"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        ›
      </a>
    </li>
    <li
      className="item   item--lastPage"
    >
      <a
        aria-label="Last page"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        »
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly in full size with max pages less than total pages 1`] = `
.c0 {
  display: inline-block;
  padding: 16px 30px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 30px;
  cursor: point;
  background-color: #f04257;
  background: #f04257;
  color: #FFFFFF;
}

.c0:hover {
  background: #00beca;
}

<div
  className=""
>
  <ul
    className="list"
  >
    <li
      className="item   item--firstPage"
    >
      <a
        aria-label="First page"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        «
      </a>
    </li>
    <li
      className="item   item--previousPage"
    >
      <a
        aria-label="Previous page"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        ‹
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 1"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        1
      </a>
    </li>
    <li
      className="item  item--disabled item--page"
    >
      <span
        className="link"
      >
        2
      </span>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 3"
        className="link  c0"
        color="red"
        href="example.com/page/3"
        onClick={[Function]}
        target="_self"
      >
        3
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 4"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        4
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 5"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        5
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 6"
        className="link  c0"
        color="red"
        href="example.com/page/6"
        onClick={[Function]}
        target="_self"
      >
        6
      </a>
    </li>
    <li
      className="item   item--nextPage"
    >
      <a
        aria-label="Next page"
        className="link  c0"
        color="red"
        href="example.com/page/3"
        onClick={[Function]}
        target="_self"
      >
        ›
      </a>
    </li>
    <li
      className="item   item--lastPage"
    >
      <a
        aria-label="Last page"
        className="link  c0"
        color="red"
        href="example.com/page/10"
        onClick={[Function]}
        target="_self"
      >
        »
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly in full size with max pages more than total pages 1`] = `
.c0 {
  display: inline-block;
  padding: 16px 30px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 30px;
  cursor: point;
  background-color: #f04257;
  background: #f04257;
  color: #FFFFFF;
}

.c0:hover {
  background: #00beca;
}

<div
  className=""
>
  <ul
    className="list"
  >
    <li
      className="item   item--firstPage"
    >
      <a
        aria-label="First page"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        «
      </a>
    </li>
    <li
      className="item   item--previousPage"
    >
      <a
        aria-label="Previous page"
        className="link  c0"
        color="red"
        href="example.com/page/3"
        onClick={[Function]}
        target="_self"
      >
        ‹
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 1"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        1
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 2"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        2
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 3"
        className="link  c0"
        color="red"
        href="example.com/page/3"
        onClick={[Function]}
        target="_self"
      >
        3
      </a>
    </li>
    <li
      className="item  item--disabled item--page"
    >
      <span
        className="link"
      >
        4
      </span>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 5"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        5
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 6"
        className="link  c0"
        color="red"
        href="example.com/page/6"
        onClick={[Function]}
        target="_self"
      >
        6
      </a>
    </li>
    <li
      className="item   item--nextPage"
    >
      <a
        aria-label="Next page"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        ›
      </a>
    </li>
    <li
      className="item   item--lastPage"
    >
      <a
        aria-label="Last page"
        className="link  c0"
        color="red"
        href="example.com/page/6"
        onClick={[Function]}
        target="_self"
      >
        »
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly in minimalist form 1`] = `
.c0 {
  display: inline-block;
  padding: 16px 30px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 30px;
  cursor: point;
  background-color: #f04257;
  background: #f04257;
  color: #FFFFFF;
}

.c0:hover {
  background: #00beca;
}

<div
  className=""
>
  <ul
    className="list"
  >
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 1"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        1
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 2"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        2
      </a>
    </li>
    <li
      className="item  item--disabled item--page"
    >
      <span
        className="link"
      >
        3
      </span>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 4"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        4
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 5"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        5
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly with custom labels 1`] = `
.c0 {
  display: inline-block;
  padding: 16px 30px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 30px;
  cursor: point;
  background-color: #f04257;
  background: #f04257;
  color: #FFFFFF;
}

.c0:hover {
  background: #00beca;
}

<div
  className=""
>
  <ul
    className="list"
  >
    <li
      className="item   item--firstPage"
    >
      <a
        aria-label="First page"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        First
      </a>
    </li>
    <li
      className="item   item--previousPage"
    >
      <a
        aria-label="Previous page"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        Previous
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 1"
        className="link  c0"
        color="red"
        href="example.com/page/1"
        onClick={[Function]}
        target="_self"
      >
        Page 1
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 2"
        className="link  c0"
        color="red"
        href="example.com/page/2"
        onClick={[Function]}
        target="_self"
      >
        Page 2
      </a>
    </li>
    <li
      className="item  item--disabled item--page"
    >
      <span
        className="link"
      >
        Page 3
      </span>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 4"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        Page 4
      </a>
    </li>
    <li
      className="item   item--page"
    >
      <a
        aria-label="Page 5"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        Page 5
      </a>
    </li>
    <li
      className="item   item--nextPage"
    >
      <a
        aria-label="Next page"
        className="link  c0"
        color="red"
        href="example.com/page/4"
        onClick={[Function]}
        target="_self"
      >
        Next
      </a>
    </li>
    <li
      className="item   item--lastPage"
    >
      <a
        aria-label="Last page"
        className="link  c0"
        color="red"
        href="example.com/page/5"
        onClick={[Function]}
        target="_self"
      >
        Last
      </a>
    </li>
  </ul>
</div>
`;
